### PR TITLE
Added logic to traverse new app model and find all dependencies

### PR DESCRIPTION
**Pull request by vijaynalawade**
Added traversing logic which is model independent. It will gather all dependencies(ref value) from application json.

Used below app for testing.
```json
{ 
   "name":"test",
   "type":"flogo:app",
   "version":"0.0.1",
   "appModel":"1.0.0",
   "description":"My flogo application description",
   "triggers":[ 
      { 
         "id":"my_rest_trigger",
         "ref":"github.com/TIBCOSoftware/flogo-contrib/trigger/rest",
         "settings":{ 
            "port":"9233"
         },
         "handlers":[ 
            { 
               "action": { 
                   "ref":"github.com/TIBCOSoftware/flogo-contrib/action/flow",
                   "data":{ 
                          "flowURI":"res://flow:myflow"
                    },
                   "mappings":{ 
                       "input":[ 
                        ],
                       "output":[ 
                       ]
                   }
                },
               "settings":{ 
                  "method":"GET",
                  "path":"/test"
               }
            }
         ]
      }
   ],
   "resources":[ 
      { 
         "id":"flow:myflow",
         "data": {
            "name": "My Flow",
            "description": "Example description",
            "metadata":{ 
               "input":[ 
                  { "name":"customerId", "type":"string" },
                  { "name":"orderId", "type":"string" }
               ],
               "output":[ 
                  { "name":"value", "type":"string" }
               ]
            },
            "tasks":[ 
               { 
                  "id":"FirstLog",
                  "type":"iterator",
                  "settings":{ 
                     "iterate":"10"
                  },
                  "activity":{ 
                     "ref":"github.com/TIBCOSoftware/flogo-contrib/activity/log",
                     "output":{ },
                     "mappings":{ 
                        "input":[ 
                           { 
                              "type":"assign",
                              "value":"$current.iteration.key",
                              "mapTo":"message"
                           }
                        ]
                     }
                  }
               },
               { 
                  "id":"SecondLog",
                  "activity":{ 
                     "ref":"github.com/TIBCOSoftware/flogo-contrib/activity/log",
                     "input":{ 
                        "message": "test message"
                     },
                     "output":{ },
                     "mappings":{ }
                  }
               }
            ],
            "links":[ 
               { 
                  "from":"FirstLog",
                  "to":"SecondLog",
                  "type":"expression",
                  "value":"$flow.orderId > 1000"
               }
            ]
         },
         "errorHandler": {
           "tasks": [
             { 
                "id":"SecondLog",
                "activity":{ 
                "ref":"github.com/TIBCOSoftware/flogo-contrib/activity/log",
                "input":{ 
                  "message": "log in error handler"
                },
               "output":{ },
               "mappings":{ }
             }
           }
         ]
       }
     }
   ]
}
```

### CODE REVIEW

File: app/api.go [hunk 1]
```diff
@@ -111,7 +111,13 @@ func doCreate(enviro env.Project, appJson, rootDir, appName, vendorDir, constrai
 	}
 
 	// Create initial files
-	deps := config.ExtractDependencies(descriptor)
+	var deps []*config.Dependency
+	if descriptor.AppModel != "" {
```
Comment (mellistibco) [inline app/api.go]: can we check version? for example, if >= 1.0. what if someone or something adds app model version to the old file?

File: config/config.go [hunk 2]
```diff
@@ -160,6 +165,49 @@ func ExtractDependencies(descriptor *FlogoAppDescriptor) []*Dependency {
...
 }
 
+func ExtractAllDependencies(appJson string) ([]*Dependency) {
+
+	dh := &depHolder{}
+
+	var descriptor interface{}
+
+	//Should be valid app json
+	json.Unmarshal([]byte(appJson), &descriptor)
```
Comment (mellistibco) [inline config/config.go]: can we catch any error here?

Comment (vijaynalawade) [reply config/config.go]: @mellistibco: The App json is already parsed in earlier step. Any json related issues will be caught there.

File: app/api.go (hunk 1 above)
Comment (vijaynalawade) [reply app/api.go]: @mellistibco: Sure. Will add specific check.

**Pull request merged**